### PR TITLE
Hangup Custom Headers

### PR DESCRIPTION
**Pull request by mmujic**
Hello,

I've added support for custom headers to hangup message, it can be used to differentiate explicit hangup.

Please note I've already signed CLA.

Thanks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ cmdline.h
 version.c
 docs/html/
 
+.idea
```
Comment (lminiero) [inline .gitignore]: Typo? Please remove.

File: plugins/janus_sip.c [hunk 2]
```diff
@@ -3240,7 +3241,9 @@ static void *janus_sip_handler(void *data) {
 			session->media.ready = FALSE;
 			session->media.on_hold = FALSE;
 			janus_sip_call_update_status(session, janus_sip_call_status_closing);
-			nua_bye(session->stack->s_nh_i, TAG_END());
+			char custom_headers[2048];
+			janus_sip_parse_custom_headers(root, (char *)&custom_headers);
+			nua_bye(session->stack->s_nh_i, TAG_IF(strlen(custom_headers) > 0, SIPTAG_HEADER_STR(custom_headers)), TAG_END());
```
Comment (lminiero) [inline plugins/janus_sip.c]: Line a bit too long, better to split in multiple lines for readability (see `nua_invite` lines for instance).

File: .gitignore (hunk 1 above)
Comment (mmujic) [reply .gitignore]: It's not a typo, CLion editor internally stores project config in .idea directory

Comment (lminiero) [reply .gitignore]: Then please move it at the end of the file, and add a comment specifying what it's for (see OSX line).

**Comment by lminiero** [comment]
Thanks, merging! :+1:

**Pull request merged**